.github/workflows/test-go.yml

```diff
@@ -207,10 +228,12 @@ jobs:
           COMMIT_SHA: ${{ github.sha }}
         run: |
           set -exo pipefail
-
+ 
           # Build the dynamically generated source files.
           make prep
-
+         
+          packages=$(echo "${{ toJSON(needs.test-matrix.outputs.matrix) }}" | jq -c -r --arg id "${{ matrix.id }}" '.[$id] | .packages') 
```

Comment:
Oh, one other thought that occurred to me. This is possibly an oversight from previous changes, but I wonder if we should assert the length of packages and fail if it's empty? Might be good to do it here while we're making this change.